tests/tests/test_demo_artifact.py

```diff
@@ -141,10 +140,18 @@ def demo_artifact_upload(self, run_demo_script, exit_cond="Login password:"):
             logger.error(str(arts))
             raise
         assert "mender-demo-artifact" in arts[0]["name"]
+
         # Bring down the demo script
+        logger.info("-- Terminating demo script")
         proc.send_signal(signal.SIGTERM)
+        # Continue logging stdout until process stops
+        for line in proc.stdout:
+            logger.info(line.decode())
         proc.wait()
-        assert proc.returncode == 0
+        stderr = proc.stderr.read().decode()
```

Comment:
`stderr` is not opened [here](https://github.com/mendersoftware/integration/blob/f899a6bcd2b3d9d4a0a345acb6e2e3bafd4f634f/tests/tests/test_demo_artifact.py#L77).

Reply:
Right, but setting this is tricky. What if it fills the OS pipe buffer and keeps the subprocess in a blocked state?